### PR TITLE
feat(filter) Adds code to allow filters to match objects with missing…

### PR DESCRIPTION
**Pull request by giorgosera**
… and/or empty values

### CODE REVIEW

File: lib/filter.js [hunk 1]
```diff
@@ -11,12 +11,45 @@ exports.parseFilter = (filter) => {
     throw new Error('Invalid filter operator');
   }
 
+  if (filter.hasOwnProperty('matchMissing') && !(typeof filter.matchMissing === 'boolean')) {
```
Comment (stelios313) [inline lib/filter.js]: can we coerse to boolean with !! to avoid this check?

Comment (giorgosera) [inline lib/filter.js]: I tried to find a way to do that. Any suggestions? Given that we can do it for any possible input such as `'hello'`, `'true'`, `1`, `undefined` etc

Comment (giorgosera) [inline lib/filter.js]: There you go: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Boolean :)

Comment (giorgosera) [inline lib/filter.js]: Actually, according to [docs](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Boolean)

> Do not use a Boolean object to convert a non-boolean value to a boolean value. Instead, use Boolean as a function to perform this task:

**Pull request merged**